File: MidtermLabWork3/mp12_table_display.js
// Import the file system module
const fs = require("fs");

// Import readline so we can ask user input
const readline = require("readline");

// Create input interface
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Ask user where the dataset is located
rl.question("Enter the CSV file path: ", function(path) {

    // Read the file
    fs.readFile(path, "utf8", function(err, data) {

        if (err) {
            console.log("Error reading the file.");
            console.log(err.message);
            rl.close();
            return;
        }

        console.log("\nDataset Table Output:\n");

        // Split rows using new line
        const rows = data.split("\n");

        // Loop through rows
        for (let i = 0; i < rows.length; i++) {

            const columns = rows[i].split(",");

            let output = "";

            // Format table columns
            for (let j = 0; j < columns.length; j++) {
                output += "| " + columns[j].padEnd(20) + " ";
            }

            output += "|";

            console.log(output);
        }

        rl.close();
    });

});

File: MidtermLabWork3/MP12TableDisplay.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Scanner;

/*
 * MP12 - Display Dataset in Formatted Table Output
 *
 * This program reads a CSV dataset file provided by the user.
 * The program parses each row of the dataset and displays the
 * contents in a formatted table structure in the console.
 */

    public class MP12TableDisplay {

        public static void main(String[] args) {

            /*
            * Variable: input
            * Scanner object used to receive user input from the keyboard.
            * In this case, the user will provide the file path of the dataset.
            */
            Scanner input = new Scanner(System.in);

            /*
            * Ask the user to input the dataset file path.
            * The program cannot continue processing until the dataset location is provided.
            */
            System.out.print("Enter CSV file path: ");
            
            /*
            * Variable: filePath
            * Stores the location of the dataset entered by the user.
            * Example: /Users/name/Documents/dataset.csv
            */
            String filePath = input.nextLine();

            try {
                
                /*
                * Dataset Handling:
                * FileReader opens the CSV dataset file.
                * BufferedReader allows the program to read the file line by line
                * instead of loading the entire dataset into memory.
                */
                BufferedReader reader = new BufferedReader(new FileReader(filePath));
                String line;

                /* Print table top border */
                System.out.println("+----------------------+----------------------+----------------------+");
                while ((line = reader.readLine()) != null) {

                    /* Split CSV row */
                    String[] columns = line.split(",");

                    /* Print table row */
                    for (int i = 0; i < columns.length; i++) {
                        System.out.printf("| %-20s ", columns[i]);
                    }

                    System.out.println("|");

                    /* Print row border */
                    System.out.println("+----------------------+----------------------+----------------------+");
                }

                reader.close();
            
            } catch (IOException e) {
                System.out.println("Error reading the file.");
            }

            input.close();
        }
    }